### PR TITLE
Allow Symfony 7

### PR DESCRIPTION
**Pull request by evertharmeling**
Follows #435

### CODE REVIEW

**Comment by evertharmeling** [comment]
Hmm looks like with fixing the `phpstan` errors it introduced `php-cs-fixer` errors...?

**Comment by bytehead** [comment]
We will do a new major anyway to drop PHP 7 support and clean up a little bit 👍

**Comment by evertharmeling** [comment]
CI runs for Symfony `6.4`/`7.0` should be running when new versions are released (probably in a few days), or we should refactor the used 'stability'.

Not sure how to fix the error in the `lowest` run...

**Comment by bytehead** [comment]
> Not sure how to fix the error in the lowest run...

Seems like the `phpunit-brige` is somewhat not compatible with the doctrine libs...

**Comment by evertharmeling** [comment]
Found what triggered it, https://github.com/symfony/phpunit-bridge/blob/v6.0.0/bootstrap.php#L33

Which is fixed in `v6.0.17` (https://github.com/symfony/phpunit-bridge/blob/v6.0.17/bootstrap.php#L33)

So added a conflict to skip versions before `6.0.17`.

File: composer.json [hunk 1]
```diff
@@ -48,12 +48,12 @@
...
-        "friendsofphp/php-cs-fixer": "3.22.*",
+        "friendsofphp/php-cs-fixer": "^3.40",
         "knplabs/gaufrette": "^0.9",
+        "m2mtech/flysystem-stream-wrapper": "^1.0",
         "oneup/flysystem-bundle": "^4.1",
         "phpstan/phpstan": "^1.8",
         "phpunit/phpunit": "^9.5",
-        "m2mtech/flysystem-stream-wrapper": "^1.0",
         "symfony/browser-kit": "^4.4 || ^5.4 || ^6.0 || ^7.0",
         "symfony/phpunit-bridge": "^6.0 || ^7.0",
```
Comment (bytehead) [inline composer.json:58]:
```suggestion
        "symfony/phpunit-bridge": "^6.0.17 || ^7.0",
```

File: composer.json [hunk 2]
```diff
@@ -65,6 +65,9 @@
         "oneup/flysystem-bundle": "^3.0",
         "m2mtech/flysystem-stream-wrapper": "^1.0 (Required when using Flysystem)"
     },
+    "conflict": {
+        "symfony/phpunit-bridge": "<6.0.17"
```
Comment (bytehead) [inline composer.json:69]: You can directly require the minimum `^6.0.17`.

**Review by bytehead — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (evertharmeling) [reply composer.json]: Right, much better!

File: src/Templating/Helper/UploaderHelper.php [hunk 3]
```diff
@@ -9,20 +9,8 @@
 
 class UploaderHelper extends Helper
```
Comment (ndench) [inline src/Templating/Helper/UploaderHelper.php:10]: The Templating component is [deprecated in Symfony 6.0](https://github.com/symfony/symfony/blob/aeb248933cd1a31b96531e1999cd97159bf0c493/UPGRADE-7.0.md#templating) and will be removed in 7.0.

I think this UploadHelper will need to change a bit more, though I'm unsure how. Do we even need to extend it? All it does is provide a charset.

I opened #437 to track this.

**Review by ndench — COMMENTED** [review]
Nice work!

File: composer.json [hunk 4]
```diff
@@ -32,33 +32,32 @@
         }
     ],
     "require": {
-        "php": "^7.4 || ^8.0",
-        "symfony/asset": "^4.4 || ^5.4 || ^6.0",
```
Comment (bytehead) [inline composer.json:36]: We can drop the support for `symfony/* ^4.4` as 4.4 is not compatible with PHP 8 anyway.

**Review by bytehead — CHANGES_REQUESTED** [review]

**Comment by bytehead** [comment]
[The pipline error](https://github.com/1up-lab/OneupUploaderBundle/actions/runs/7530303418/job/20496377432?pr=436) seems somewhat unrelated...

**Comment by evertharmeling** [comment]
> [The pipline error](https://github.com/1up-lab/OneupUploaderBundle/actions/runs/7530303418/job/20496377432?pr=436) seems somewhat unrelated...

Yes, it's a weird one...

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -1,80 +1,93 @@
...
-                  extensions: dom, fileinfo, filter, gd, hash, intl, json, mbstring, pcre, pdo_mysql, zlib
-                  coverage: none
+      - name: Checkout
+        uses: actions/checkout@v3
 
-            - name: Checkout
-              uses: actions/checkout@v2
+      - name: Install the dependencies
+        run: |
+          composer require symfony/framework-bundle:^${{ matrix.symfony }} symfony/http-foundation:^${{ matrix.symfony }}
```
Comment (bytehead) [inline .github/workflows/ci.yml:69]:
```suggestion
          composer require symfony/framework-bundle:^${{ matrix.symfony }} symfony/http-foundation:^${{ matrix.symfony }} symfony/yaml:^${{ matrix.symfony }}
```

maybe this fixes the unit test 😬

**Review by bytehead — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 5 above)
Comment (evertharmeling) [reply .github/workflows/ci.yml]: It did, nice!

**Review by bytehead — APPROVED** [review]
Thank you big times 🙏 🥳

**Pull request merged**